pybaseball/statcast.py

```diff
@@ -10,20 +11,22 @@
 
 _SC_SINGLE_GAME_REQUEST = "/statcast_search/csv?all=true&type=details&game_pk={game_pk}"
 _SC_SMALL_REQUEST = "/statcast_search/csv?all=true&hfPT=&hfAB=&hfBBT=&hfPR=&hfZ=&stadium=&hfBBL=&hfNewZones=&hfGT=R%7CPO%7CS%7C=&hfSea=&hfSit=&player_type=pitcher&hfOuts=&opponent=&pitcher_throws=&batter_stands=&hfSA=&game_date_gt={start_dt}&game_date_lt={end_dt}&team={team}&position=&hfRO=&home_road=&hfFlag=&metric_1=&hfInn=&min_pitches=0&min_results=0&group_by=name&sort_col=pitches&player_event_sort=h_launch_speed&sort_order=desc&min_abs=0&type=details&"
+DATE_FORMAT = "%Y-%m-%d"
 
-def validate_datestring(date_text):
+def validate_datestring(date_text: Optional[str]) -> datetime.datetime:
     try:
-        datetime.datetime.strptime(date_text, '%Y-%m-%d')
-    except ValueError:
+        assert date_text
+        return datetime.datetime.strptime(date_text, DATE_FORMAT)
+    except (AssertionError, ValueError):
         raise ValueError("Incorrect data format, should be YYYY-MM-DD")
 
-def sanitize_input(start_dt, end_dt):
+def sanitize_input(start_dt: Optional[str], end_dt: Optional[str]) -> Tuple[datetime.datetime, datetime.datetime]:
     # If no dates are supplied, assume they want yesterday's data
     # send a warning in case they wanted to specify
     if start_dt is None and end_dt is None:
         today = datetime.datetime.today()
-        start_dt = (today - datetime.timedelta(1)).strftime("%Y-%m-%d")
-        end_dt = today.strftime("%Y-%m-%d")
+        start_dt = (today - datetime.timedelta(1)).strftime(DATE_FORMAT)
```

Comment:
If we use `timedelta(days=1)` then we still have a *date* type, not a datetime type, and then just `str(dt)` gives us what we need, not the strftime

Reply:
`datetime.datetime.today()` is a datetime type, but you raise a lot of good points here that really point out that these shouldn't be datetimes, since we don't care about the time. They should be dates.

Reply:
`datetime.date.today()` is a date type, not a datetime type. Use that instead